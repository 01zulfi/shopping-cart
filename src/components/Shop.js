import React from "react";

const Shop = () => {
  const item = (asteroid) => (
    <div key={asteroid.id} className="item-div" data-id={asteroid.id}></div>
  );

  return <div className="shop">{item({ id: 1 })}</div>;
};

export default Shop;
