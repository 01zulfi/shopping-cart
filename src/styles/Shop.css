div.shop {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
}

.shop .item-div {
  width: 90%;
  box-shadow: 0px 15px 30px rgb(0 0 0 / 15%);
  background-color: rgba(255, 255, 255, 0.09);
  border: 1px solid #d4bff9;
  margin-top: 20px;
  border-radius: 10px;
  margin-bottom: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 15px;
}

.shop .item-div img {
  width: 85%;
}

.shop .item-div h2 {
  background-color: rgba(255, 255, 255, 0.11);
  width: 85%;
  margin-top: 5px;
  margin-bottom: 5px;
  text-align: center;
}

.shop .item-div p {
  font-size: 20px;
}

.shop .item-div button {
  animation: button-animation 3s infinite linear alternate;
  padding: 10px;
  border: 1px solid #d4bff9;
  color: black;
  border: 0;
  font-size: 19px;
  border-radius: 5px;
  margin: 15px;
}

@keyframes button-animation {
  from {
    background-color: #bc75c9;
  }
  to {
    background-color: #ab47bc;
  }
}

.shop .item-div button:hover {
  cursor: pointer;
  font-weight: bold;
}

@media (min-width: 768px) {
  div.shop {
    width: 85%;
    flex-direction: row;
    flex-wrap: wrap;
    align-items: stretch;
    justify-content: center;
  }

  .shop .item-div {
    width: 26%;
    margin: 25px;
    justify-content: space-between;
  }
}
