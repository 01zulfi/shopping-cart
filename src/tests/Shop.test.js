import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Shop from "../components/Shop";

test("shop renders", () => {
  const { container } = render(<Shop />);
  expect(container.firstChild).toMatchSnapshot();
  expect(container.firstChild).toHaveClass("shop");
});

test("items render", () => {
  const { container } = render(<Shop />);
  const item = container.querySelectorAll(".item-div");
  expect(item).toHaveLength(7);
});

test("first asteroid renders correct info", () => {
  const { container } = render(<Shop />);
  const firstName = container.querySelector(".asteroid-name");
  const firstPrice = container.querySelector(".asteroid-price");

  expect(firstName).toHaveTextContent("1 Ceres");
  expect(firstPrice).toHaveTextContent("253");
});
