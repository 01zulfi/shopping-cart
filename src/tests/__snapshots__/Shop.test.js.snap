// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shop renders 1`] = `
<div
  class="shop"
>
  <div
    class="item-div"
    data-id="1482816745855"
  >
    <hr />
    <p
      class="asteroid-name"
    >
      Name : 
      1 Ceres
    </p>
    <p
      class="asteroid-price"
    >
      Price: 
      253
    </p>
    <hr />
  </div>
  <div
    class="item-div"
    data-id="68402194543"
  >
    <hr />
    <p
      class="asteroid-name"
    >
      Name : 
      4 Vesta
    </p>
    <p
      class="asteroid-price"
    >
      Price: 
      299
    </p>
    <hr />
  </div>
  <div
    class="item-div"
    data-id="911312589980"
  >
    <hr />
    <p
      class="asteroid-name"
    >
      Name : 
      25143 Itokawa
    </p>
    <p
      class="asteroid-price"
    >
      Price: 
      99
    </p>
    <hr />
  </div>
  <div
    class="item-div"
    data-id="814780917834"
  >
    <hr />
    <p
      class="asteroid-name"
    >
      Name : 
      243 Ida
    </p>
    <p
      class="asteroid-price"
    >
      Price: 
      147
    </p>
    <hr />
  </div>
  <div
    class="item-div"
    data-id="718603216526"
  >
    <hr />
    <p
      class="asteroid-name"
    >
      Name : 
      433 Eros
    </p>
    <p
      class="asteroid-price"
    >
      Price: 
      444
    </p>
    <hr />
  </div>
  <div
    class="item-div"
    data-id="1265177198944"
  >
    <hr />
    <p
      class="asteroid-name"
    >
      Name : 
      253 Mathilde
    </p>
    <p
      class="asteroid-price"
    >
      Price: 
      121
    </p>
    <hr />
  </div>
  <div
    class="item-div"
    data-id="1525632476903"
  >
    <hr />
    <p
      class="asteroid-name"
    >
      Name : 
      101955 Bennu
    </p>
    <p
      class="asteroid-price"
    >
      Price: 
      191
    </p>
    <hr />
  </div>
</div>
`;
